integrations/telegram/src/misc/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { axios } from '@botpress/client'
+import { PICTURE_LIMIT_SIZE } from 'src/const'
+import { Telegraf } from 'telegraf'
+import { PhotoSize, User } from 'telegraf/typings/core/types/typegram'
+
+export const getUserNameFromTelegramUser = (telegramUser: User) => {
+  if (telegramUser.last_name) {
+    return `${telegramUser.first_name} ${telegramUser.last_name}`
+  } else if (telegramUser.username) {
+    return `${telegramUser.first_name} (username: ${telegramUser.username})`
```

Comment:
This concatenation in this specific line might be too long for display purposes? Also, having the `username` available guarantees the `first_name` is available? Maybe we could use the following fallback sequence (in order of preference, I think) in this function for simplicity?

`first_name + last_name` OR `first_name` OR `username` OR `Telegram user ${telegramUser.id}`

Reply:
first_name is supposed to always be available, however I thought that firstName only would not make much sense, that's why I added username as well. Also first_name + last_name can be extremely long in some cases too.

Reply:
Good point about first + last name being possibly too long, that being considered, what would you think of the following approach? (once we agree and polish it later after testing we may put it in a common package so the logic is shared across integrations):

```ts
if (telegramUser.last_name && telegramUser.last_name.charCodeAt(0) < 256) {
  return `${telegramUser.first_name} ${telegramUser.last_name.toLocaleUpperCase().charAt(0)}.`
} else if (telegramUser.username) {
  return telegramUser.username
}
return `Telegram user ${telegramUser.id}`
```

This basically shows the first name plus the abbreviate last name only if there's a last name and it can be reasonably assumed it's a western last name (as we cannot do the same for non-western names that use other alphabets). If not then it just prints the username if available or just a generic labeled user ID as fallback.

Reply:
I'd rather store the full lastName and delegate the display to the frontend. Also I think it's a pity not to store the firstName if it is available, what do you think ?